### PR TITLE
feat: add --remove-source and --skip-ts options, fix duplicate import bug

### PR DESCRIPTION
**Pull request by jchip**
### 🗑️ `--remove-source` Option
- **Purpose**: Automatically delete original source files after successful conversion
- **Safety**: Only removes files after confirmed successful conversion
- **Respects**: `--dry-run` mode (won't delete in dry-run)
- **Error Handling**: Graceful failure with warnings if deletion fails
- **Usage**: `ts2mjs --cjs lib/ --remove-source`

### 🚫 `--skip-ts` Option  
- **Purpose**: Skip all TypeScript files (.ts and .d.ts) during conversion
- **Backwards Compatibility**: Default behavior unchanged (still converts .ts files)
- **Use Case**: When you only want to convert .js files, not TypeScript files
- **Usage**: `ts2mjs --cjs lib/ --skip-ts`

### 🐛 Duplicate Import Handling
- **Issue**: When a file imported the same module multiple times, only the first occurrence was converted
- **Root Cause**: Require regex pattern was too restrictive, requiring semicolon immediately after require()
- **Fix**: Relaxed regex pattern to match require() calls in any context
- **Impact**: Now correctly handles cases like `__exportStar(require("./module.js"), exports)`

### CLI Interface
```bash
# New options added to help output
--remove-source       Remove the original files after successful conversion
--skip-ts            Skip all TypeScript files (.ts and .d.ts) instead of converting them
```

### Code Changes
- **processFiles.ts**: Added `removeSource` and `skipTs` to Options interface
- **app.ts**: Added CLI option parsing and file deletion logic  
- **processFile.ts**: Updated file type checking and regex patterns
- **Comprehensive test coverage**: All new features fully tested

### Remove Source Files
```bash
# Convert and remove original files
ts2mjs --cjs dist/ --remove-source

# Safe dry-run (won't delete anything)
ts2mjs --cjs dist/ --remove-source --dry-run
```

### Skip TypeScript Files
```bash
# Only convert .js files, skip .ts/.d.ts files
ts2mjs --cjs lib/ --skip-ts

# Default behavior (converts .ts files)
ts2mjs --cjs lib/
```

## Testing
- ✅ **73 tests passing** (including new comprehensive test cases)
- ✅ **Backwards compatibility maintained**
- ✅ **Manual testing with real-world TypeScript compiler output**
- ✅ **Edge cases covered** (duplicate imports, complex file structures)

## Breaking Changes
**None** - All changes are additive and backwards compatible.

### CODE REVIEW

**Comment by Jason3S** [comment]
@jchip,

Wow, thank you. I'll take a look.

File: src/processFile.ts [hunk 1]
```diff
@@ -75,21 +75,17 @@ export function processSourceFile(src: SourceFile, options: ProcessFileOptions):
...
     let linesChanged = 0;
-    let match: RegExpExecArray | null;
-    while ((match = exp.exec(content))) {
-        const { index } = match;
+    const content = magicString.toString();
 
-        const line = match[0];
-        const reference = match.groups?.['file'];
+    // Use replace with a callback to handle all matches at once
+    const processedContent = content.replace(new RegExp(regExpImportExport, 'g'), (match, ...args) => {
```
Comment (Jason3S) [inline src/processFile.ts]:
```suggestion
    const processedContent = content.replaceAll(new RegExp(regExpImportExport), (match, ...args) => {
```

No need for the extra `g` since it is already in `regExpImportExport` and we will need `d`.

File: src/processFile.ts [hunk 2]
```diff
@@ -8,11 +8,11 @@ import { readSourceFile, SOURCE_MAP_URL_MARKER } from './readSourceFile.js';
 import type { SourceFile, SourceMap } from './SourceFile.js';
 import { UsageError } from './errors.js';
 
-const isSupportedFileMJS = /\.(m?js|d\.m?ts)$/;
-const isSupportedFileCJS = /\.(c?js|d\.c?ts)$/;
+const isSupportedFileMJS = /\.(m?js|m?ts|d\.m?ts)$/;
+const isSupportedFileCJS = /\.(c?js|c?ts|d\.c?ts)$/;
 
 const regExpImportExport = /(import|export).*? from ('|")(?<file>\..*?)\2;/g;
```
Comment (Jason3S) [inline src/processFile.ts]:
```suggestion
const regExpImportExport = /(import|export).*? from ('|")(?<file>\..*?)\2;/dg;
```
I wasn't aware of the `d` flag when this code was written. Using it will make things easier.

File: src/processFile.ts [hunk 3]
```diff
@@ -8,11 +8,11 @@ import { readSourceFile, SOURCE_MAP_URL_MARKER } from './readSourceFile.js';
...
-const regExpRequire = /require\(('|")(?<file>\..*?)\1\);/g;
+const regExpRequire = /require\(('|")(?<file>\..*?)\1\)/g;
```
Comment (Jason3S) [inline src/processFile.ts]:
```suggestion
const regExpRequire = /require\(('|")(?<file>\..*?)\1\)/dg;
```

File: src/processFile.ts [hunk 4]
```diff
@@ -75,21 +75,17 @@ export function processSourceFile(src: SourceFile, options: ProcessFileOptions):
...
-        const { index } = match;
+    const content = magicString.toString();
 
-        const line = match[0];
-        const reference = match.groups?.['file'];
+    // Use replace with a callback to handle all matches at once
+    const processedContent = content.replace(new RegExp(regExpImportExport, 'g'), (match, ...args) => {
+        // Extract groups from the match - the groups are at the end of args array
+        const groups = args[args.length - 1];
+        const reference = groups?.['file'];
```
Comment (Jason3S) [inline src/processFile.ts:92]:
```suggestion
        const reference = groups?.['file'];
        const pos = match.indices?.groups?.['file'];
```

Extract the position of the match.

File: src/processFile.ts [hunk 5]
```diff
@@ -75,21 +75,17 @@ export function processSourceFile(src: SourceFile, options: ProcessFileOptions):
...
 
         if (reference && isRelativePath(reference)) {
```
Comment (Jason3S) [inline src/processFile.ts]:
```suggestion
        if (reference && pos && isRelativePath(reference)) {
            const [start, end] = pos;
```

File: src/processFile.ts [hunk 6]
```diff
@@ -102,32 +98,34 @@ function adjustImportExportStatements(src: SourceFile, options: ProcessFileOptio
                 }
             }
             const newRef = calcRelativeImportFilename(reference, srcFilename, srcRoot, targetRoot, ext);
```
Comment (Jason3S) [inline src/processFile.ts:108]:
```suggestion
            const newRef = calcRelativeImportFilename(reference, srcFilename, srcRoot, targetRoot, ext);
            magicString.update(start, end, newRef);
```

Add back the `magicString.update`

File: src/processFile.ts [hunk 7]
```diff
@@ -102,32 +98,34 @@ function adjustImportExportStatements(src: SourceFile, options: ProcessFileOptio
...
-            continue;
+            return match.replace(reference, newRef);
         }
+        return match;
+    });
+
+    // Apply all changes at once
+    if (linesChanged > 0) {
+        magicString.update(0, content.length, processedContent);
     }
```
Comment (Jason3S) [inline src/processFile.ts]:
```suggestion
```

Applying everything at the end breaks the .map file. The whole point of using magicString is to be able to click on some compiled code and have the editor take you to the source.

File: src/processFile.ts [hunk 8]
```diff
@@ -102,32 +98,34 @@ function adjustImportExportStatements(src: SourceFile, options: ProcessFileOptio
...
+
     return linesChanged;
 }
 
 function adjustRequireStatements(src: SourceFile, options: ProcessFileOptions, magicString: MagicString) {
```
Comment (Jason3S) [inline src/processFile.ts:120]: The comments above apply to this function too.

File: src/processFile.ts [hunk 9]
```diff
@@ -140,16 +138,29 @@ function adjustRequireStatements(src: SourceFile, options: ProcessFileOptions, m
...
     return linesChanged;
 }
 
-export function isSupportedFileType(filename: string, ext: ExtensionType): boolean {
-    return (ext === '.mjs' ? isSupportedFileMJS : isSupportedFileCJS).test(filename);
+export function isSupportedFileType(filename: string, ext: ExtensionType, skipTs = false): boolean {
+    const baseSupported = (ext === '.mjs' ? isSupportedFileMJS : isSupportedFileCJS).test(filename);
+
+    // If skipTs is enabled, exclude ALL .ts files (including .d.ts files)
+    if (skipTs && /\.ts$/.test(filename)) {
```
Comment (Jason3S) [inline src/processFile.ts]:
```suggestion
    if (skipTs && /\.[cm]?ts$/.test(filename)) {
```

**Review by Jason3S — COMMENTED** [review]
Thank you very much.

I have added some requested changes to improve the resulting .map files.

File: fyn-lock.yaml
Comment (Jason3S) [inline fyn-lock.yaml:1]: Please remove this file and add `fyn-lock.*` to `.gitignore`.

**Comment by Jason3S** [comment]
@jchip,

Thank you. PR merged with #345.
